src/libs/actions/BankAccounts.js

```diff
@@ -821,6 +829,59 @@ function updateReimbursementAccountDraft(bankAccountData) {
 }
 
 /**
+<<<<<<< HEAD
+ * Triggers a modal to open allowing the user to reset their bank account
+ */
+function requestResetFreePlanBankAccount() {
+    Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {shouldShowResetModal: true});
+}
+
+/**
+ * Hides modal allowing the user to reset their bank account
+ */
+function cancelResetFreePlanBankAccount() {
+    Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {shouldShowResetModal: false});
+}
+
+/**
+ * Reset user's reimbursement account. This will delete the bank account.
+ *
+ * @param {String} password
+ */
+function resetFreePlanBankAccount(password = 'Password1') {
+    const bankAccountID = lodashGet(reimbursementAccountInSetup, 'bankAccountID');
+    if (!bankAccountID) {
+        throw new Error('Missing bankAccountID when attempting to reset free plan bank account');
+    }
+    if (!credentials || !credentials.login) {
+        throw new Error('Missing credentials when attempting to reset free plan bank account');
+    }
+    if (!password) {
+        throw new Error('Must provide password to delete bank account');
+    }
+
+    const previousACHData = {...reimbursementAccountInSetup};
+    Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {achData: null, shouldShowResetModal: false});
+    API.DeleteBankAccount({bankAccountID, ownerEmail: credentials.login, password})
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                // Unable to delete bank account so we restore the bank account details
```

Comment:
Just curious why we would wipe the data and restore it vs just wipe it once the API call succeeds?

Reply:
I wonder if this can be made clearer somehow.. but basically the idea is that we

- optimistically assume the request will succeed (UI changes fast)
- if we succeed then the UI remains correct
- if we fail then we will set the app state back to whatever it was before

I pretty much have just assumed we would want a similar behavior here, but maybe it's not correct to do so. In our README for the app we have this at the moment:

![2021-10-21_13-58-23](https://user-images.githubusercontent.com/32969087/138372620-f492679a-265a-4b4e-8485-0404f229dcef.png)

So I applied that same concept to deleting... what do you think? Does it make sense?

Reply:
Added a comment above!